### ISSUE TITLE
Overflow ellipsis broken on file paths in breadcrumbs

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#13535]
https://sourcegraph.com/github.com/cssnano/cssnano@master/-/blob/packages/postcss-normalize-display-values/src/lib/map.js

![image](https://user-images.githubusercontent.com/10532611/91845397-5e815480-ec59-11ea-8c06-bbcef5b7daea.png)

Long path segments no longer get truncated with ellipsis.
We should also add a story for this.

**Comment by tjkandala** [comment]
This happens because [hyphens sugggest line-break opportunities](https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens#Suggesting_line_break_opportunities). Add `white-space: nowrap;` to fix.

**Comment by felixfbecker** [comment]
Awesome investigation @tjkandala! I propose we make that change as a quick fix, as the current state looks very broken.

> I don't think the breadcrumbs and action items should be on the same row. Especially now that we've added 'Home > Repositories >', even medium-length file paths are pushing action items out of view.

We need to be concious about vertical screen estate though since it subtracts estate from the code content.  If we put them on two rows, we would have a lot of whitespace under the file path. @AlicjaSuska's designs had them wrap in two rows when the truncation becomes very small, maybe we can do a hybrid approach where things are truncated to a certain degree, but only to a certain `min-width` per segment (i.e. no single-letter truncated path segments) after which it will wrap to two rows. Wdyt?

> Scrolling > truncation for long file paths. If a user has multiple directories with the same prefix and filenames, truncating the names could lead to confusion.

Not sure I agree with this, because horizontal scrolling is generally a very difficult UI for mouse users, especially on macOS where scrollbars are hidden by default. The problem with prefixes is true, but it's a trade-off between hiding different parts of the path: truncation favors showing the beginning of each path segment but showing the full path, while scrolling favors showing the beginning of the path, while hiding the end of the path by default. I personally think showing a bit of every path segment works better in most situations. Curious how @sourcegraph/product-team thinks about this.

**Comment by tjkandala** [comment]
Thanks! I hesitated to fix it ASAP because I noticed that the segments weren't being truncated at a certain point, pushing the action items out of view. See this picture on Sourcegraph.com 
<img width="997" alt="Screen Shot 2020-09-21 at 10 52 33 AM" src="https://user-images.githubusercontent.com/37420160/93782457-95171300-fbf8-11ea-87cf-deb7667e9b7a.png">

However, this would look much better than the current state, so I'll merge.
